src/internal/storage/chunk/metadata.go

```diff
@@ -66,7 +66,7 @@ type Entry struct {
 // IT HAS BEEN USED IN A RELEASED MIGRATION
 func SetupPostgresStoreV0(ctx context.Context, tx *pachsql.Tx) error {
 	_, err := tx.Exec(`
-	CREATE TABLE storage.chunk_objects (
+	CREATE TABLE IF NOT EXISTS storage.chunk_objects (
```

Comment:
Changes to the schema here makes it easier to call migrations in a test since these operations will now be idempotent.

Reply:
This seems like an issue with how migrations and tests are run. Shouldn’t every test be run against a particular state of the database?

I think that this is ignoring a broken state when a migration is run twice.

Reply:
I want to be able to call `fileset.NewTestStorage()` and set up the storage tables, then I wanted to run the migrations to just add the new tables and schemas. This seemed like the path of least resistance. The other alternative would be to write an alternative `fileset.NewTestStorage()` that manually creates just the tables that aren't created by the migrations.

Reply:
I agree with @robert-uhl that we shouldn't be modifying the already released SQL, and in general we shouldn't use conditionals in the schema SQL.

A couple of thoughts about how to manage the database during tests:
- We should have one database for each test.  That might mean test constructors have to take db as a parameter.
- We should run all the migrations once per test at the beginning.  I had originally wanted to run portions of the migrations, but that's more complicated and the migrations package doesn't make it very easy to do that.
- clusterstate should continue to be at the bottom of the dependency graph, so any subsystem can import it to setup the entire database for "unit" tests.


Separately: I think we want to initialize the entire storage layer for the test, not just the fileset API.  I think we need a way to setup the server from `internal/storage`.